chain-extensions/xvm/src/lib.rs

```diff
@@ -87,7 +93,25 @@ where
                 // Similar to EVM behavior, the `source` should be (limited to) the
                 // contract address. Otherwise contracts would be able to do arbitrary
                 // things on behalf of the caller via XVM.
-                let source = env.ext().address();
+                let source = env.ext().address().clone();
+
+                // Claim the default evm address if needed.
+                let mut actual_weight = Weight::zero();
```

Comment:
Maybe I'm mistaken, but the `UA::to_h160` also consumes gas/weight - it's a DB read.

Reply:
Updated.